src/xml/writer.cr

```diff
@@ -0,0 +1,312 @@
+    end
+
+    # ditto
+    def attributes(attributes : Hash | NamedTuple)
+      attributes.each do |key, value|
+        attribute key.to_s, value
+      end
+    end
+
+    # Emits text content.
+    #
+    # Text content can happen inside of an element, attribute value, cdata, dtd, etc.
+    def text(content : String) : Nil
+      call WriteString, content
+    end
+
+    # Emits the start of a CDATA section.
+    def start_cdata : Nil
+      call StartCDATA
+    end
+
+    # Emits the end of a CDATA section.
+    def end_cdata : Nil
+      call EndCDATA
+    end
+
+    # Emits the start of a CDATA section, invokes the block
+    # and then emits the enf of the CDATA section.
+    def cdata
+      start_cdata
+      yield.tap { end_cdata }
+    end
+
+    # Emits a CDATA section.
+    def cdata(text : String) : Nil
+      call WriteCDATA, text
+    end
+
+    # Emits the start of a comment.
+    def start_comment : Nil
+      call StartComment
+    end
+
+    # Emits the end of a comment.
+    def end_comment : Nil
+      call EndComment
+    end
+
+    # Emits the start of a comment, invokes the block
+    # and the emits the end of the comment.
+    def comment
+      start_comment
+      yield.tap { end_comment }
+    end
+
+    # Emits a comment.
+    def comment(text : String) : Nil
+      call WriteComment, text
+    end
+
+    # Emits the start of a DTD.
+    def start_dtd(name : String, pubid : String, sysid : String) : Nil
+      call StartDTD, name, pubid, sysid
+    end
+
+    # Emits the end of a DTD.
+    def end_dtd : Nil
+      call EndDTD
+    end
+
+    # Emits the start of a DTD, invokes the block
+    # and then emits the end of the DTD.
+    def dtd(name : String, pubid : String, sysid : String) : Nil
+      start_dtd name, pubid, sysid
+      yield.tap { end_dtd }
+    end
+
+    # Emits a DTD.
+    def dtd(name : String, pubid : String, sysid : String, subset : String? = nil) : Nil
+      call WriteDTD, name, pubid, sysid, string_to_unsafe(subset)
+    end
+
+    # Emits a namespace.
+    def namespace(prefix, uri)
+      attribute prefix, "xmlns", nil, uri
+    end
+
+    # Forces content written to this writer to be flushed to
+    # this writer's IO.
+    def flush
+      call Flush
+    end
+
+    # Sets the indent string.
+    def indent=(str : String)
+      if str.empty?
+        call SetIndent, 0
+      else
+        call SetIndent, 1
+        call SetIndentString, str
+      end
+    end
+
+    # Sets the indent level (number of spaces)
+    def indent=(level : Int)
+      if level <= 0
+        call SetIndent, 0
+      else
+        call SetIndent, 1
+        call SetIndentString, " " * level
+      end
+    end
+
+    # Set the quote char to use, either ' or ".
+    def quote_char=(char : Char)
+      unless char == '\'' || char == '"'
+        raise ArgumentError.new("quote char must be ' or \", not #{char}")
+      end
+
+      call SetQuoteChar, char.ord
+    end
+
+    # TODO: mark as private
```

Comment:
Last thing I knew, the binaries that travis uses for linux weren't up to date. I'll try to change this and see if it's already fixed :-)